### ISSUE TITLE
Add import support to `okta_profile_mapping`

### ISSUE THREAD

**Issue by exitcode0** [okta/terraform-provider-okta#1955]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<hr>

### Description

Currently the documentation states that there is "no reason" to import `okta_profile_mapping`

I disagree with this perspective and think that this resource should support import

Adding import support to this resource will make it easier to bring existing configuration under TF management
It means that the terraform plan can be used to align your HCl with the current state of the resource regardless of how you craft your HCL

<hr>

### New or Affected Resource(s)

- okta_profile_mapping

<hr>

### References 

- #888 

<hr>

### Other information
This is a duplicate of #888 I figured I should raise a new issue rather than necro-post on the old issue, if you feel differently we can close this one and re-open the original

**Comment by duytiennguyen-okta** [comment]
OKTA internal reference https://oktainc.atlassian.net/browse/OKTA-717456

**Comment by duytiennguyen-okta** [comment]
That statement was made 4 years ago and probably not true anymore

**Comment by duytiennguyen-okta** [comment]
@exitcode0 I have to revert this because of the weird behaviour surrounding [d.GetOK() ](https://github.com/okta/terraform-provider-okta/pull/2032)

**Comment by exitcode0** [comment]
@duytiennguyen-okta did the importer not work without the problematic changes to `d.GetOK()`?

**Comment by duytiennguyen-okta** [comment]
It is related to delete_when_absent, which impact mappings. I will need to convert from sdk-plugin to plugin-framework to addess the issue

**Comment by exitcode0** [comment]
Is that the only way to work around this?
I'm happy to contribute a PR to get this back into the provider if there is a simpler MVP fix that we can do here, perhaps we can lift inspiration from another resource in the provider?

**Comment by duytiennguyen-okta** [comment]
I don't know there is another way, since GetOK() get confused false with null for delete_when_absent, which lead to all kind of issue. The plugin-frameworks will be able to differentiate between these 2 values

**Comment by exitcode0** [comment]
I'm not super familiar with provider development yet, something i'd like to improve so I can contribute more here

Do we have access to `d.GetOkExists()` easily in this file?

**Comment by duytiennguyen-okta** [comment]
Yeah. Here is the thread with the same problem that we currently having https://discuss.hashicorp.com/t/terraform-sdk-usage-which-out-of-get-getok-getokexists-with-boolean/41815/2

**Comment by aditya-okta** [comment]
@exitcode0 , is this required or can we close the issue?

**Comment by exitcode0** [comment]
> [@exitcode0](https://github.com/exitcode0) , is this required or can we close the issue?

I think that import support for this resource is still required
that said, i've not taken a look at the code recently to check if this is now properly supported

**Comment by dhiwakar-okta** [comment]
Hi @exitcode0 

I've created PR https://github.com/okta/terraform-provider-okta/pull/2809 to address this.

Thanks
Dhiwakar
